modules/tinymce/src/themes/silver/main/ts/ui/dialog/IFrame.ts

```diff
@@ -25,7 +25,35 @@ const getDynamicSource = (initialData: Optional<string>): IFrameSourcing => {
       // TINY-3769: We need to use srcdoc here, instead of src with a data URI, otherwise browsers won't retain the Origin.
       // See https://bugs.chromium.org/p/chromium/issues/detail?id=58999#c11
       if (cachedValue.get() !== html) {
-        Attribute.set(frameComponent.element, 'srcdoc', html);
+        const iframeElement = frameComponent.element as SugarElement<HTMLIFrameElement>;
+        const setSrcdocValue = () => Attribute.set(iframeElement, 'srcdoc', html);
+
+        if (stream) {
+          const iframe = iframeElement.dom;
+          const iframeDoc = Optional.from(iframe.contentDocument);
+          const iframeWindow = Optional.from(iframe.contentWindow);
+
+          const isScrollAtBottom = () => {
+            const isElementScrollAtBottom = ({ scrollTop, scrollHeight, clientHeight }: HTMLElement) => scrollTop + clientHeight >= scrollHeight;
+            return iframeDoc.map((doc) => doc.documentElement).map((docEl) => isElementScrollAtBottom(docEl)).getOr(false);
+          };
+          const isAtBottom = isScrollAtBottom();
+
+          iframeDoc.fold(
+            setSrcdocValue,
+            (doc) => {
+              doc.open();
+              doc.write(html);
+              doc.close();
+            }
+          );
+
+          if (isAtBottom) {
+            iframeDoc.map((doc) => doc.body).each((body) => iframeWindow.each((win) => win.scrollTo(0, body.scrollHeight)));
```

Comment:
`lift2` might make this easier, we used to have other "do things with two optionals" methods but surprisingly it doesn't come up much within the editor itself.
https://github.com/tinymce/tinymce/blob/ba6994f39b09ecd2c28cef68c3e8a85a4815e235/modules/katamari/src/main/ts/ephox/katamari/api/Optionals.ts#L62-L63

Reply:
Resolved as part of https://github.com/tinymce/tinymce/pull/8868#discussion_r1251369006